### PR TITLE
Add Carthage support

### PR DESCRIPTION
**Pull request by diogoguimaraes**
Adds Carthage support 😃

### CODE REVIEW

File: Leash.xcodeproj/project.xcworkspace/contents.xcworkspacedata [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (LucianoPolit) [inline Leash.xcodeproj/project.xcworkspace/contents.xcworkspacedata]: Can you delete this file? It should not be in the repo.

File: Leash.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (LucianoPolit) [inline Leash.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist]: Can you delete this file? It should not be in the repo.

File: Cartfile [hunk 3]
```diff
@@ -0,0 +1,2 @@
+github "Alamofire/Alamofire" == 4.8.2
```
Comment (LucianoPolit) [inline Cartfile]: Can you change it to `~> 4.5` and `~> 4.0`?

File: Leash.podspec [hunk 4]
```diff
@@ -15,15 +15,15 @@ Pod::Spec.new do |s|
 
   s.subspec 'Core' do |ss|
     ss.source_files  = 'Source/Core/**/*.swift'
-    ss.dependency      'Alamofire', '~> 4.5'
+    ss.dependency      'Alamofire', '~> 4.8.2'
```
Comment (LucianoPolit) [inline Leash.podspec]: `~> 4.8` here.

File: Leash.podspec [hunk 5]
```diff
@@ -15,15 +15,15 @@ Pod::Spec.new do |s|
...
   end
 
   s.subspec 'RxSwift' do |ss|
     ss.source_files  = 'Source/RxSwift/*.swift'
     ss.dependency      'RxSwift', '~> 4.0'
```
Comment (LucianoPolit) [inline Leash.podspec]: `~> 5.0` here.

File: Package.swift [hunk 6]
```diff
@@ -1,8 +0,0 @@
-import PackageDescription
```
Comment (LucianoPolit) [inline Package.swift:1]: You made me remember that I wanted to delete this file since SPM can not be used for iOS yet. I'll add it again in the future.

**Review by LucianoPolit — CHANGES_REQUESTED** [review]

**Comment by LucianoPolit** [comment]
Thanks for taking your time, good work mate! I just left you a few comments, everything works fine though 👍

**Comment by LucianoPolit** [comment]
One thing to mention is that I do not want to do a major version bump until `Alamofire 5.0` is released. That's the reason why I proposed to maintain the same dependencies. This way, a minor version increment would be the only requirement.

File: Cartfile.private [hunk 7]
```diff
@@ -0,0 +1 @@
+github "AliSoftware/OHHTTPStubs" "8.0.0"
```
Comment (LucianoPolit) [inline Cartfile.private]: `~> 6.0` here.

File: Leash.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -0,0 +1,771 @@
...
+				LD_RUNPATH_SEARCH_PATHS = (
+					"$(inherited)",
+					"@executable_path/Frameworks",
+					"@loader_path/Frameworks",
+				);
+				PRODUCT_BUNDLE_IDENTIFIER = org.Tests;
+				PRODUCT_NAME = "$(TARGET_NAME)";
+				SWIFT_OBJC_BRIDGING_HEADER = "";
+				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
+				SWIFT_VERSION = 5.0;
```
Comment (LucianoPolit) [inline Leash.xcodeproj/project.pbxproj]: Also, would you mind changing this version to 4.0 (from Xcode)?

File: Leash.podspec [hunk 9]
```diff
@@ -15,15 +15,15 @@ Pod::Spec.new do |s|
 
   s.subspec 'Core' do |ss|
     ss.source_files  = 'Source/Core/**/*.swift'
-    ss.dependency      'Alamofire', '~> 4.5'
```
Comment (LucianoPolit) [inline Leash.podspec]: Let's undo the changes of this file

File: Cartfile.private (hunk 7 above)
Comment (diogoguimaraes) [reply Cartfile.private]: I've changed it to `~> 7.0` because it was when they updated to Swift 4.2, otherwise I will not be able to build the project with a Swift 5 project. Is that ok?

**Review by LucianoPolit — APPROVED** [review]

File: Cartfile.private (hunk 7 above)
Comment (LucianoPolit) [reply Cartfile.private]: :+1:
